### ISSUE TITLE
Spack containerize doesn't strip local repo's

### ISSUE THREAD

**Issue by GuillaumeQuenneville** [LLNL/spack#28263]
### Steps to reproduce

Create a spack.yaml with a local repo

```
spack:
  specs:
  - patchelf
  - package-in-local-repo
  concretization: together
  view: true
  repos:
  - ./spack-repo
  container:
    format: docker
    images: 
      os: ubuntu:18.04
      spack: v0.17.1
    strip: true
```

spack containerize will not strip the repo from the config and will crash when a package from the repo that doesn't exist .

My first intuition is to solve this one of these ways
- copy the local repo onto the build image to create the build (i.e. with COPY ./spack-repo /opt/spack-environment/spack-repo before the "spack env activate && spack install" stage)
- make dockerfile work with local repo. Not sure how this would work since you can't load a volume when building an image
- warn user that local repo will be ignored and strip it from the spack.yaml in the dockerfile

Thanks!

### Error message
```
 => ERROR [builder 4/6] RUN cd /opt/spack-environment &&     spack env activate . &&     spack install --fail-fast &&     spack gc -y                                                                                                         1.3s
------
 > [builder 4/6] RUN cd /opt/spack-environment &&     spack env activate . &&     spack install --fail-fast &&     spack gc -y:
#10 1.197 ==> Warning: Failed to initialize repository: './spack-repo'.
#10 1.197   No repo.yaml found in './spack-repo'
#10 1.197   To remove the bad repository, run this command:
#10 1.197       spack repo rm ./spack-repo
#10 1.197 ==> Error: Package 'zofu' not found.
```

### Information on your system

* **Spack:** 0.17.1
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-zen2
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
There's a plan to implement support for custom repositories in https://github.com/spack/spack/discussions/20442 with details on how we want to do that. The idea is to generate a context (i.e. a directory) with all the necessary file and COPY them in the image.

**Comment by GuillaumeQuenneville** [comment]
Thanks! I only checked issues, forgot about discussions...

**Comment by alalazo** [comment]
Need to verify, but I think this is solved by #35187

**Comment by alalazo** [comment]
Fixed by #35187 With the customization of the template, a user can setup the environment to `COPY` the custom repository to the correct path in the Docker image.
